client/src/test/java/com/paypal/selion/android/sample/AppiumAndroidCheckBoxSeekBarTest.java

```diff
@@ -32,58 +32,54 @@
-    private static final String TEXT_VIEW_LOCATOR = "com.paypal.selion.pageobjectsdemoapp:id/seekBar_textview";
+    private static final String ACTION_BUTTON_LOCATOR = "com.paypal.selion.pageobjectsdemoapp:id/btnNext";
+    private static final String SEEK_BAR_LOCATOR = "com.paypal.selion.pageobjectsdemoapp:id/stateSeekBar";
+    private static final String TEXT_VIEW_LOCATOR = "com.paypal.selion.pageobjectsdemoapp:id/stateTextSlider";
     // Check box
-    private static final String CHECKBOX_ANDROID_LOCATOR = "com.paypal.selion.pageobjectsdemoapp:id/android_checkbox";
-    private static final String CHECKBOX_IOS_LOCATOR = "com.paypal.selion.pageobjectsdemoapp:id/ios_checkbox";
-    private static final String VALUE_TEXTVIEW_LOCATOR = "com.paypal.selion.pageobjectsdemoapp:id/checkbox_textview";
+    private static final String CHECKBOX_LOCATOR = "com.paypal.selion.pageobjectsdemoapp:id/stateSwitch";
+    private static final String VALUE_TEXTVIEW_LOCATOR = "com.paypal.selion.pageobjectsdemoapp:id/stateTxtSwitch";
 
     private UiButton menuButton;
     private UiObject seekBar;
     private UiTextView seekBarTextView;
     private UiTextView checkBoxTextView;
-    private UiObject iosCheckBox;
-    private UiObject androidCheckBox;
+    private UiObject checkBox;
 
     @BeforeClass
     public void initElements() {
         menuButton = new UiButton(ACTION_BUTTON_LOCATOR);
         seekBar = new UiObject(SEEK_BAR_LOCATOR);
         seekBarTextView = new UiTextView(TEXT_VIEW_LOCATOR);
-        iosCheckBox = new UiObject(CHECKBOX_IOS_LOCATOR);
-        androidCheckBox = new UiObject(CHECKBOX_ANDROID_LOCATOR);
+        checkBox = new UiObject(CHECKBOX_LOCATOR);
         checkBoxTextView = new UiTextView(VALUE_TEXTVIEW_LOCATOR);
     }
 
     @Test
     @MobileTest(appPath = PAGE_OBJECTS_APP_PATH, device = DEVICE_NAME)
     public void testCheckBox() throws InterruptedException {
         WebDriverWaitUtils.waitUntilElementIsVisible(ACTION_BUTTON_LOCATOR);
-        menuButton.click(LONG_PRESS_BUTTON_LOCATOR);
-        WebDriverWaitUtils.waitUntilElementIsVisible(LONG_PRESS_BUTTON_LOCATOR);
-        menuButton.click(androidCheckBox);
-        androidCheckBox.click();
-        Assert.assertEquals(checkBoxTextView.getText(), "Android");
-        iosCheckBox.click();
-        Assert.assertEquals(checkBoxTextView.getText(), "iOS");
-        iosCheckBox.click();
+        menuButton.click(ACTION_BUTTON_LOCATOR);
+        menuButton.click(ACTION_BUTTON_LOCATOR);
+        menuButton.click(checkBox);
+        checkBox.click();
+        Assert.assertEquals(checkBoxTextView.getText(), "Switch is OFF");
+        checkBox.click();
+        Assert.assertEquals(checkBoxTextView.getText(), "Switch is ON");
+        checkBox.click();
     }
 
     @Test
     @MobileTest(appPath = PAGE_OBJECTS_APP_PATH, device = DEVICE_NAME)
     public void testSeekBar() throws InterruptedException {
```

Comment:
@alizelzele This test now consistently fails for me when I run the suite `AppiumAndroidPageObjectsSuite.xml`  .. I'm using Android 4.4.4 API 19